### PR TITLE
Feature: #102337 - PSR-14 event for modifying record list export data

### PR DESCRIPTION
**Pull request by linawolf**
References: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/934
Releases: main

### CODE REVIEW

File: Documentation/ApiOverview/Events/Events/Backend/_BeforeRecordDownloadIsExecutedEvent/_DataListener.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        }
+
+        // Redact actual content...
+        foreach ($records as $uid => $record) {
+            foreach ($gdprFields as $gdprField) {
+                if (isset($record[$gdprField])) {
+                    $records[$uid][$gdprField] = '(REDACTED)';
+                }
+            }
+        }
```
Comment (sfroemkenjw) [inline Documentation/ApiOverview/Events/Events/Backend/_BeforeRecordDownloadIsExecutedEvent/_DataListener.php:35]:
```suggestion
        foreach ($records as $rowKey => $record) {
            foreach ($gdprFields as $gdprField) {
                if (isset($record[$gdprField])) {
                    $records[$rowKey][$gdprField] = '(REDACTED)';
                }
            }
        }
```

Comment (sfroemkenjw) [reply Documentation/ApiOverview/Events/Events/Backend/_BeforeRecordDownloadIsExecutedEvent/_DataListener.php:35]: In Core each line will be added via `$records[]`, so there is no record UID available

**Review by sfroemkenjw — COMMENTED** [review]
I don't understand the FrontendLogin modifications in that patch.

File: Documentation/ApiOverview/Events/Events/Backend/_BeforeRecordDownloadIsExecutedEvent/_DataListener.php (hunk 1 above)
Comment (linawolf) [reply Documentation/ApiOverview/Events/Events/Backend/_BeforeRecordDownloadIsExecutedEvent/_DataListener.php]: Done, thanks for catching it

**Comment by linawolf** [comment]
> I don't understand the FrontendLogin modifications in that patch.

I updated the code snippets and there had been core chnages since the last code snippet update

**Review by sfroemkenjw — APPROVED** [review]

**Pull request merged**